### PR TITLE
Header Component

### LINKED ISSUE #33: [AMA] Create File structure
# What?
For the AMA, we need to create a "cloud storage" like file structure. You'll need to be able to go through the folder and dig deeper.

# Keep in mind
- There are a couple of types of resources, but for now, we don't need a different icon per file type, it's a nice to have
- Make your components as atomic as possible
- For UI guidelines, see the Figma file: https://www.figma.com/design/Db518nvwhA0tiQQa47e7pr/AMA-Prototype?node-id=97-2&t=3P3mBFDlGaCD82v4-1

### LINKED ISSUE #38: [AMA] Create Header Component

### PR DESCRIPTION
**Pull request by BT-Creator**
Header component with the following structure:
- [X] Base component
- [X] Tab component
- [ ] Context Menu component

Closes #38

### CODE REVIEW

File: loama/src/components/Header.vue [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import Loama from '../assets/loama.svg'
+import { store } from '@/store';
+import router from '@/router';
+const props = defineProps<{
+    pfpSrc: string,
+
+}>()
+
+const isContextMenuOpen = ref(false)
+console.log(store.usedPod)
```
Comment (Zelzahn) [inline loama/src/components/Header.vue]: :eyes:

File: loama/src/components/Header.vue [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    h1 {
+        font-family: "JetBrains Mono";
+        font-size: calc(var(--base-unit)*6);
+        font-style: normal;
+        font-weight: 700;
+        line-height: normal;
+        color: var(--solid-purple);
+        text-transform: uppercase;
+    }
```
Comment (Zelzahn) [inline loama/src/components/Header.vue]: This should ideally be set global right?

File: loama/src/components/Header.vue [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        right: 0;
+        color: var(--off-white);
+    }
+    
+    .visible {
+        display: block;
+    }
+</style>
+
+<script setup lang="ts">
```
Comment (Zelzahn) [inline loama/src/components/Header.vue]: `<style>` cannot be before `<script>` (fyi order of `<script>` vs. `<template>` may be freely chosen)

https://vuejs.org/style-guide/rules-recommended.html#single-file-component-top-level-element-order

File: loama/src/components/HeaderTab.vue [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    min-width: calc(var(--base-unit)*15);
+}
+
+.active {
+    background-color: var(--solid-purple);
+    color: var(--off-white);
+}
+</style>
+
+<script setup lang="ts">
```
Comment (Zelzahn) [inline loama/src/components/HeaderTab.vue]: Same as two comments ago

File: loama/src/components/Header.vue (hunk 1 above)
Comment (BT-Creator) [reply loama/src/components/Header.vue]: Was still developing, but I'll remove it before we merge

File: loama/src/components/Header.vue (hunk 2 above)
Comment (BT-Creator) [reply loama/src/components/Header.vue]: I'll be quite honest: Maybe? It's more for the wordmark, but you can also argue sementically that the loama text in the header is not part of the main content as an article and shouldn't be an `h1`.

I might replace it with a `<span>` to be more sementically correct

File: loama/src/components/Header.vue (hunk 3 above)
Comment (BT-Creator) [reply loama/src/components/Header.vue]: Alright, didn't know that, keep that in mind

**Review by Zelzahn — APPROVED** [review]
LGTM

**Pull request merged**